src/main/java/net/dries007/tfc/objects/te/TEBarrel.java

```diff
@@ -79,6 +79,14 @@ public TEBarrel()
     public void readFromItemTag(NBTTagCompound nbt)
     {
         tank.readFromNBT(nbt.getCompoundTag("tank"));
+        if (tank.getFluidAmount() > tank.getCapacity())
+        {
+            // Fix config changes
+            FluidStack fluidStack = tank.getFluid();
+            //noinspection ConstantConditions
+            fluidStack.amount = tank.getCapacity();
+            tank.setFluid(fluidStack);
+        }
```

Comment:
Is this preventing a crash? Or trying to do some sort of data-fixer change for configuration changes. If the latter, I don't think we should, as it's going to be inconsistent (already I see crucibles just ignore possible changes). It's reasonable to assume that if you change a configuration option, it won't affect previous items / blocks destructively.

Reply:
It's the latter, mostly because fluid render overflows in GUI. I will do like crucibles then